configure.ac

```diff
@@ -211,49 +190,13 @@ AC_CHECK_FUNCS([ \
 AX_PTHREAD(, [AC_MSG_ERROR([Cannot find pthreads])])
 CC="${PTHREAD_CC}"
 
-AC_ARG_VAR([LIBGPG_ERROR_CFLAGS], [C compiler flags for libgpg-error])
-AC_ARG_VAR([LIBGPG_ERROR_LIBS], [linker flags for libgpg-error])
-if test -z "${LIBGPG_ERROR_LIBS}"; then
-	AC_MSG_CHECKING([for libgpg-error])
-	if ! test -x "${with_libgpg_error_prefix}/bin/gpg-error-config"; then
-		AC_MSG_ERROR([Cannot locate libgpg-error])
-	else
-		AC_MSG_RESULT([found])
-		LIBGPG_ERROR_CFLAGS="`\"${with_libgpg_error_prefix}/bin/gpg-error-config\" --cflags`"
-		LIBGPG_ERROR_LIBS="`\"${with_libgpg_error_prefix}/bin/gpg-error-config\" --libs`"
-	fi
-fi
-
-AC_ARG_VAR([LIBASSUAN_CFLAGS], [C compiler flags for libassuan])
-AC_ARG_VAR([LIBASSUAN_LIBS], [linker flags for libassuan])
-if test -z "${LIBASSUAN_LIBS}"; then
-	AC_MSG_CHECKING([for libassuan])
-	test -x "${with_libassuan_prefix}/bin/libassuan-config" || AC_MSG_ERROR([Cannot locate libassuan])
-	"${with_libassuan_prefix}/bin/libassuan-config" --version | grep "^2\." > /dev/null || AC_MSG_ERROR([Need assuan-2])
-
-	AC_MSG_RESULT([found])
-
-	LIBASSUAN_CFLAGS="`\"${with_libassuan_prefix}/bin/libassuan-config\" --cflags`"
-	LIBASSUAN_LIBS="`\"${with_libassuan_prefix}/bin/libassuan-config\" --libs`"
-fi
-
-AC_ARG_VAR([LIBGCRYPT_CFLAGS], [C compiler flags for libgcrypt])
-AC_ARG_VAR([LIBGCRYPT_LIBS], [linker flags for libgcrypt])
-if test -z "${LIBGCRYPT_LIBS}"; then
-	AC_MSG_CHECKING([for libgcrypt])
-	if ! test -x "${with_libgcrypt_prefix}/bin/libgcrypt-config"; then
-		AC_MSG_ERROR([Cannot locate libgcrypt])
-	else
-		AC_MSG_RESULT([found])
-		LIBGCRYPT_CFLAGS="`\"${with_libgcrypt_prefix}/bin/libgcrypt-config\" --cflags`"
-		LIBGCRYPT_LIBS="`\"${with_libgcrypt_prefix}/bin/libgcrypt-config\" --libs`"
-	fi
-fi
-
+PKG_CHECK_MODULES([LIBGPG_ERROR], [gpg-error], [HAVE_LIBGPG_ERROR="yes"])
+PKG_CHECK_MODULES([LIBASSUAN], [libassuan], [HAVE_LIBASSUAN="yes"])
+PKG_CHECK_MODULES([LIBGCRYPT], [libgcrypt], [HAVE_LIBGCRYPT="yes"])
```

Comment:
these are not optional libraries, we do not need the `HAVE_*` variables, we should just fail.

Reply:
I've modified commit to make those three libraries unconditional.

BTW if gnupg-pkcs11-scd is build with openssl libcrypto it should be possible to use this API instead libgcrypto.
Am I right? 🤔